bodyswap.lua

```diff
@@ -75,13 +75,64 @@ function createNemesis(unit)
   return nemesis
 end
 
-function updateAdvCompanionsList(advHistFig)
-  df.global.ui_advmode.companions.all_histfigs:resize(0)
-  for _,link in ipairs(advHistFig.histfig_links) do
-    if link._type == df.histfig_hf_link_companionst then
-      df.global.ui_advmode.companions.all_histfigs:insert('#',link.target_hf) -- we only need to update the historical figure ID list as the other stuff here is reset whenever the companions screen is loaded, and is probably not used for anything else
+function isPet(nemesis)
+  if nemesis.unit then
+    if nemesis.unit.relationship_ids.Pet ~= -1 then
+      return true
+    end
+  elseif nemesis.figure then -- in case the unit is offloaded
+    for  _, link in ipairs(nemesis.figure.histfig_links) do
+      if link._type == df.histfig_hf_link_pet_ownerst then
+        return true
+      end
+    end
+  end
+  return false
+end
+
+function processNemesisParty(nemesis, targetUnitID, alreadyProcessed)
+-- configures the target and any leaders/companions to behave as cohesive adventure mode party members
+  local alreadyProcessed = alreadyProcessed or {}
+  alreadyProcessed[tostring(nemesis.id)] = true
+
+  local nemUnit = nemesis.unit
+  if nemesis.unit_id == targetUnitID then -- the target you're bodyswapping into
+    df.global.ui_advmode.interactions.party_core_members:insert('#', nemesis.figure.id)
```

Comment:
I don't have an adventure save handy to test this: do you have any idea if DF expects these vectors of IDs to be sorted? If so, `utils.insert_sorted()` might be a good idea in these cases.

Reply:
My impression is that it doesn't based on the observation that the IDs in `party_extra_members` and `party_pets` are ordered according to the sequence in which the corresponding figures joined the party, though it's difficult to say for certain with `party_core_members` as additional figures aren't added to this list during normal gameplay.

Reply:
Ok! Hopefully DF is consistent, then.